### PR TITLE
Fix duplicate errors for secret notifications

### PR DESCRIPTION
**Pull request by a-roberts**
# Changes

For https://github.com/tektoncd/dashboard/issues/1144

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by a-roberts** [comment]
Noticed getSecretsErrorMessage being called twice, this is an improvement but seeing an error for the on close button click handler now and the message I do get (once, not twice) is unformatted, even though everywhere we've got an InlineNotification it's using getErrorMessage which surely gives us the nicely formatted one

File: src/containers/Secrets/Secrets.js [hunk 1]
```diff
@@ -502,8 +501,7 @@ function mapStateToProps(state, props) {
   const namespace = namespaceParam || getSelectedNamespace(state);
 
   return {
-    errorMessage:
-      getDeleteSecretsErrorMessage(state) || getSecretsErrorMessage(state),
+    errorMessage: getDeleteSecretsErrorMessage(state),
```
Comment (AlanGreene) [inline src/containers/Secrets/Secrets.js]: do we end up missing some classes of error now?

ideally we'd just have one 'error' state, and set it with the correct details depending on whether it's create, delete, etc.

Comment (a-roberts) [reply src/containers/Secrets/Secrets.js]: Yeah that's my concern too so just going through it again now (it seemed a little easy, I do think moving towards a (I think you said `global`) error state would thwart these kind of problems from repeatedly occurring)

Comment (a-roberts) [reply src/containers/Secrets/Secrets.js]: Had a good try creating secrets, switching namespaces etc, even deleting ones (shouldn't be impacted) with errors (e.g. deleting first with a kubectl command), and patching to one we just deleted - that reports a success but I'm wondering if it did anyway on master so will check

**Comment by a-roberts** [comment]
@AlanGreene can you take another look please? Didn't find any notifications no longer showing (see latest comment)

**Pull request merged**